### ISSUE TITLE
Not executing UConnect Actions

### ISSUE THREAD

**Issue by mtnears** [gyahalom/homebridge-uconnect#1]
**Describe Your Problem:**
I seem to have everything configured properly, best that I can tell :-). I get the four tiles for my Ram 3500 in the Home app.  When I look at the details it shows me the details of the vehicle including the VIN but. when I try to use the items they don't respond.  I included the responses in the log below.

**Logs:**

```
[10/11/2022, 7:28:10 AM] Homebridge v1.5.0 (HAP v0.10.4) (Homebridge 1A49) is running on port 51682.
[10/11/2022, 7:28:19 AM] [RAM UConnect] Restoring existing accessory from cache: 3500
[10/11/2022, 7:28:29 AM] [RAM UConnect] Lock failed to submit request: undefined
[10/11/2022, 7:28:31 AM] [RAM UConnect] Lock command failed
[10/11/2022, 7:28:39 AM] [RAM UConnect] Lock failed to submit request: undefined
[10/11/2022, 7:28:40 AM] [RAM UConnect] Lock command failed
[10/11/2022, 7:28:51 AM] [RAM UConnect] Engine Stop failed to submit request: undefined
[10/11/2022, 7:28:53 AM] [RAM UConnect] Engine stop command failed
```

**Plugin Config:**

```json
Show your Homebridge config.json here, remove any sensitive information.
```

**Screenshots:**
<img width="796" alt="Screen Shot 2022-10-11 at 7 38 19 AM" src="https://user-images.githubusercontent.com/115566072/195122404-daf742aa-7456-4545-8c11-b6b13f7caa3b.png">

**Environment:**
* **Car Year/Make/Model**: 2022 Ram 3500
* **Plugin Version**: 1.0.1
* **Homebridge Version**: 1.5.0
* **Node.js Version**: 16.6.0
* **NPM Version**: 8.19.2
* **Operating System**: MacOS 10.15.7

**Comment by gyahalom** [comment]
To make sure, you can successfully issue the commands from the [Mopar web app](www.mopar.com/en-us) using the same credentials and PIN?

**Comment by mtnears** [comment]
> To make sure, you can successfully issue the commands from the [Mopar web app](www.mopar.com/en-us) using the same credentials and PIN?

I can confirm it works via the Mopar (RAM) app.  Same credentials and PIN.  The logs I sent seem to show that it is accessing things fine.  I uninstalled and reconfigured just in case I missed. something and am getting the same results.   Not sure if it has to do with it being RAM and not Mopar?  But it does seem to get the details of the vehicle fine.

**Comment by gyahalom** [comment]
To be honest I don't know if the issue is due to RAM/Chrysler difference. I will try to publish an update that may give some more info in debug logs to maybe help pinpoint the issue. If you're up for it we can also try to capture some of the network traffic when you access the Mopar site from your computer to compare the requests/responses to the ones I captured and the plugin expects.

**Comment by mtnears** [comment]
Once again I'm delayed in response, sorry don't always check this.  Yea, happy to do that, I'll see if you already published. an update if not please do and I'll get it loaded and let. me know if there's anything I need to set to grab those logs.

**Comment by andyrosen** [comment]
I have the same issue with a Ram truck.  Much like mtnears, it pulls the VIN and vehicle name, so I'm certain the initial connection and authentication are working but when trying to execute an action, it gives an undefined error. 

I too am happy to help with trying to debug.   

`[10/02/2023, 06:49:37] [Uconnect] Lock failed to submit request: undefined
[10/02/2023, 06:49:39] [Uconnect] Lock command failed`

**Comment by Noldat** [comment]
I am having the same issue

2/27/2023, 1:25:29 PM] [homebridge-uconnect] Triggered SET LockTargetState: 1
[2/27/2023, 1:25:36 PM] [homebridge-uconnect] Lock failed to submit request: undefined
[2/27/2023, 1:25:37 PM] [homebridge-uconnect] Lock request status ended with: Request failed with status code 400
[2/27/2023, 1:25:37 PM] [homebridge-uconnect] Lock command failed

Ram 2500 2022

**Comment by sbels** [comment]
I have the same problem. Truck can be controlled via Ram app (or with Alexa Skill) but can't get it to work with this plugin. Does the plugin only work with older vehicles that use the Uconnect app? Mine is using the Ram app as others have mentioned.

**Comment by jeronthenet23** [comment]
I have the same issue as mtnears. My trucks both show up but commands are failing. I can issue commands successfully from the app. I use the UCONNECT app with my 2018 power wagon and my 2023 Gladiator

**Comment by gummybear9999** [comment]
I keep getting an error that all commands failed. I configure many time and also changed the code thinking that may help no luck on a 2018 jeep any idea?

**Comment by andrewmcd7** [comment]
I have the exact same issue. Any updates or anything we can do?

**Comment by vcdx71** [comment]
Another same here, 2019 RAM 1500. Works fine in the app.

[8/2/2023, 3:03:05 PM] [homebridge-uconnect] Unlock command failed
[8/2/2023, 3:03:12 PM] [homebridge-uconnect] Unlock command failed
[8/2/2023, 3:03:48 PM] [homebridge-uconnect] Lock command failed
[8/2/2023, 3:03:57 PM] [homebridge-uconnect] Lock command failed

**Comment by millhoo** [comment]
Same here.  2024 Grand Cherokee works fine in app.  None of the commands from plugin work.

11/23/2023, 9:54:15 AMUconnect BridgeERROR[Uconnect Bridge C487@On] The write handler for the characteristic 'On' on the accessory 'GRAND  CHEROKEE  LIMITED  4X4' was slow to respond!
11/23/2023, 9:54:18 AMUconnect BridgeUconnectERROREngine Start failed to submit request: undefined
11/23/2023, 9:54:20 AMUconnect BridgeUconnectERROREngine start command failed

**Comment by andrewmcd7** [comment]
Could we get an update on this? Would be soooo grateful.

**Comment by carlocardella** [comment]
Same problem with Jeep Grand Cherokee 4xe (hybrid):

```log
[4/28/2024, 8:30:19 PM] [Carlo Cardella] Engine Start failed to submit request: undefined
[4/28/2024, 8:30:20 PM] [Carlo Cardella] Engine start command failed
```

Latest homebridge, plugin and Node.js versions

**Comment by rajan60014** [comment]
Does anyone know how to use the Ram Connect Services online? I'm not sure this plugin is talking to the right API given the segmentation of the RAM Connect from UConnect? I can't figure out how to access this functionality via web. Only via the iOS or Android device.

**Comment by rajan60014** [comment]
Yeah, somehow the auth w/ Mopar is working. So it can find the truck and some of it's info. But all the functionality is hosted via ramtrucks not mopar or uconnect. So different endpoints I think. Don't know how to find the proper endpoints for RAM.
